backend/Dockerfile

```diff
@@ -14,7 +14,7 @@ WORKDIR /app
 RUN mvn clean package -DskipTests
 
 # Stage 2: Run the application
-FROM eclipse-temurin:21-jdk-jammy
+FROM build as run
```

Comment:
Minor: I made this mistake too, but the different casing on "FROM" and "as" causes warnings. Please change "as" to "AS". Instances on lines 4, 17, and 29.